diff --git a/eglot.el b/eglot.el
index f247c43..974e7d8 100644
--- a/eglot.el
+++ b/eglot.el
@@ -3685,7 +3685,15 @@ at point.  With prefix argument, prompt for ACTION-KIND."
                                     (string-prefix-p action-kind (plist-get a :kind)))
                            collect a)))
     (if interactive
-        (eglot--read-execute-code-action actions server action-kind)
+        (let ((serverName (plist-get (eglot--server-info server) :name)))
+          (if (string-equal serverName "sql-ls")
+              (with-current-buffer (get-buffer-create "*sql-ls-eglot results*")
+                (with-help-window (buffer-name)
+                  (message "In if cond")
+                  (erase-buffer)
+                  (insert (eglot--read-execute-code-action actions server action-kind))))
+            (eglot--read-execute-code-action actions server action-kind)))
+
       actions)))
 
 (defalias 'eglot-code-actions-at-mouse (eglot--mouse-call 'eglot-code-actions)
